Re-introduce likes on comments

tclaus opened the pull request:
This is the continued work initially started by @Flaburgan with PR #7918 

Current status: It works  (mostly): Likes on comments, persist to database, added tests. 
- Added like-icon on Post, Single post view and mobile view

Flaburgan committed:
Bring back likes on comments

Flaburgan committed:
Introduce like-interactions.js

Adapt to latest development

User likes
 Set css class for inline likes on comment

Re-set participation on comment likes

Co-authored-by: Thorsten Claus <ThorstenClaus@web.de>

diaspora-code-review commented on app/assets/javascripts/app/collections/likes.js:
```diff
@@ -4,7 +4,11 @@ app.collections.Likes = Backbone.Collection.extend({
+    // A comment- like has a post reference and a comment reference
+    this.url = (options.comment != null) ?
+      // not delegating to post.url() because when it is in a stream collection it delegates to that url
+      "/comments/" + options.comment.id + "/likes":
```
Infix operators must be spaced.

diaspora-code-review commented on app/assets/javascripts/app/models/comment.js:
```diff
@@ -1,6 +1,14 @@
+
+  initialize: function() {
+    this.post = this.collection.post;
+    this.interactions = new app.models.Post.LikeInteractions(_.extend({comment: this, post: this.post}, this.get("interactions")));
```
Line 8 exceeds the maximum line length of 120.

diaspora-code-review commented on app/assets/javascripts/app/models/comment.js:
```diff
@@ -1,6 +1,14 @@
+    this.post = this.collection.post;
+    this.interactions = new app.models.Post.LikeInteractions(_.extend({comment: this, post: this.post}, this.get("interactions")));
+    this.likes = this.interactions.likes;
+    this.likes_count = this.attributes.likes_count;
```
Identifier 'likes_count' is not in camel case.

diaspora-code-review commented on app/assets/javascripts/app/models/post/like_interactions.js:
```diff
@@ -0,0 +1,57 @@
+  },
+
+  userLike: function() {
+    return this.likes.select(function(like){
```
Missing space before opening brace.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
+    }
+  end
+
+  def  build_interactions_json
```
Layout/ExtraSpacing: Unnecessary spacing detected.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
+
+  def  build_interactions_json
+    {
+      likes:            as_api(likes),
```
Layout/HashAlignment: Align the keys and values of a hash literal if they span more than one line.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
+  def  build_interactions_json
+    {
+      likes:            as_api(likes),
+      likes_count:      likes_count
```
Layout/HashAlignment: Align the keys and values of a hash literal if they span more than one line.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
     mentioned_people.map {|m| PersonPresenter.new(m).as_api_json }
   end
+
+ def as_api(collection)
```
- Layout/IndentationWidth: Use 2 (not 1) spaces for indentation.
- Layout/IndentationConsistency: Inconsistent indentation detected.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
   end
+
+ def as_api(collection)
+    collection.includes(author: :profile).map {|element|
```
Layout/IndentationWidth: Use 2 (not 3) spaces for indentation.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
+    collection.includes(author: :profile).map {|element|
+      element.as_api_response(:backbone)
+    }
+  end
```
Layout/DefEndAlignment: `end` at 43, 2 is not aligned with `def` at 39, 1.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
+      element.as_api_response(:backbone)
+    }
+  end
+
```
Layout/EmptyLinesAroundClassBody: Extra empty line detected at class body end.

diaspora-code-review commented on app/services/comment_service.rb:
```diff
@@ -15,7 +15,13 @@ def find_for_post(post_id)
   def find!(comment_guid)
-    Comment.find_by!(guid: comment_guid)
+    comment = Comment.find_by!(guid: comment_guid)
+    return comment
```
Style/RedundantReturn: Redundant `return` detected.

diaspora-code-review commented on app/services/comment_service.rb:
```diff
@@ -15,7 +15,13 @@ def find_for_post(post_id)
+
+  def find_by_id!(comment_id)
+    comment = Comment.find_by!(id: comment_id)
+    return comment
```
Style/RedundantReturn: Redundant `return` detected.

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -5,11 +5,17 @@ def initialize(user=nil)
   end
 
+  def create_for_comment(comment_id)
+    comment = comment_service.find_by_id!(comment_id)
```
Rails/DynamicFindBy: Use `find_by!` instead of dynamic `find_by_id!`.

tclaus replied:
Is it Ok to have "find_by!" and "find_by_guid"?

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -5,11 +5,17 @@ def initialize(user=nil)
 
+  def create_for_comment(comment_id)
+    comment = comment_service.find_by_id!(comment_id)
+    post = post_service.find!(comment.post.id) #checks implicit for public posts
```
- Lint/UselessAssignment: Useless assignment to variable - `post`.
- Layout/LeadingCommentSpace: Missing space after `#`.

tclaus replied:
Find a post is used to check if its visible - how can this be better written?

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -25,6 +31,14 @@ def find_for_post(post_id)
   end
 
+  def find_for_comment(comment_id)
+
```
Layout/EmptyLinesAroundMethodBody: Extra empty line detected at method body beginning.

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -25,6 +31,14 @@ def find_for_post(post_id)
 
+  def find_for_comment(comment_id)
+
+    comment = comment_service.find_by_id!(comment_id)
```
Rails/DynamicFindBy: Use `find_by!` instead of dynamic `find_by_id!`.

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -25,6 +31,14 @@ def find_for_post(post_id)
+  def find_for_comment(comment_id)
+
+    comment = comment_service.find_by_id!(comment_id)
+    post = post_service.find!(comment.post.id) #checks implicit for public posts
```
- Lint/UselessAssignment: Useless assignment to variable - `post`.
- Layout/LeadingCommentSpace: Missing space after `#`.

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -36,11 +50,26 @@ def unlike_post(post_id)
   end
 
+  def unlike_comment(comment_id)
+    likes = comment_service.find_by_id!(comment_id).likes
```
Rails/DynamicFindBy: Use `find_by!` instead of dynamic `find_by_id!`.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -2,47 +2,87 @@
+
+    it "fails if comment does not exist" do
+      expect {
+         LikeService.new(alice).create_for_comment("unknown_id")
```
Layout/IndentationWidth: Use 2 (not 3) spaces for indentation.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -2,47 +2,87 @@
+      }.to raise_error ActiveRecord::RecordNotFound
+    end
+
+    it "fails if user already liked the comment" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -2,47 +2,87 @@
+      expect {
+        LikeService.new(alice).create_for_comment(bobs_comment.id)
+      }.to raise_error ActiveRecord::RecordInvalid
+    end 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -66,29 +106,49 @@
   end
 
+  describe "#destroy_comment_likes" do
+    let(:like) {LikeService.new(bob).create_for_comment(alice_comment)}
```
- Layout/SpaceInsideBlockBraces: Space missing inside {.
- Layout/SpaceInsideBlockBraces: Space missing inside }.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -66,29 +106,49 @@
+  describe "#destroy_comment_likes" do
+    let(:like) {LikeService.new(bob).create_for_comment(alice_comment)}
+
+    it "let the user destroy its own comment like" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -66,29 +106,49 @@
+      expect(result).to be_truthy
+    end
+
+    it "doesn't let the parent author destroy other comment likes" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -66,29 +106,49 @@
+    end
+
+    it "fails if the like doesn't exist" do
+      expect { 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -66,29 +106,49 @@
+
+    it "fails if the like doesn't exist" do
+      expect { 
+        result = LikeService.new(alice).destroy("unknown id")
```
Lint/UselessAssignment: Useless assignment to variable - `result`.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
   end
 
+  describe "#find_for_comment" do
+    context "with user" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+        post = alice.post(:status_message, text: "hello", public: true)
+        comment = CommentService.new(alice).create(post.id, "I like my own post")
+
+       [alice, bob, eve].map {|user| LikeService.new(user).create_for_comment(comment.id)}
```
- Layout/IndentationConsistency: Inconsistent indentation detected.
- Layout/SpaceInsideBlockBraces: Space missing inside }.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+        comment = CommentService.new(alice).create(post.id, "I like my own post")
+
+       [alice, bob, eve].map {|user| LikeService.new(user).create_for_comment(comment.id)}
+       [alice, bob, eve].each do |user|
```
Layout/IndentationConsistency: Inconsistent indentation detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+          expect(
+            LikeService.new(user).find_for_comment(comment.id).first.author.id
+          ).to be user.person.id
+        end
```
Layout/BlockAlignment: `end` at 214, 8 is not aligned with `[alice, bob, eve].each do |user|` at 210, 7.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+    end
+
+    context "without user" do
+      it "returns likes for a comment on a public post" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+        expect(LikeService.new.find_for_comment(comment.id)).to include(like)
+      end
+
+      it "doesn't return likes for a private post comment" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
       expect(post.likes.length).to eq(0)
     end
+
+    it "removes the like for a comment" do 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+
+    it "removes the like for a comment" do 
+      comment = CommentService.new(alice).create(post.id, "I like my own post")
+      
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+      expect(comment.likes.length).to eq(1)
+
+      LikeService.new(alice).unlike_comment(comment.id)
+      comment = CommentService.new(alice).find_by_id!(comment.id)
```
Rails/DynamicFindBy: Use `find_by!` instead of dynamic `find_by_id!`.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,83 @@
+      comment = CommentService.new(alice).find_by_id!(comment.id)
+      expect(comment.likes.length).to eq(0)
+    end
+
```
Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body end.

diaspora-code-review commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,26 @@ def as_api_response
+    }
+  end
+
+end
```
Layout/TrailingEmptyLines: Final newline missing.

diaspora-code-review commented on app/services/like_service.rb:
```diff
@@ -25,6 +31,13 @@ def find_for_post(post_id)
 
+  def find_for_comment(comment_id)
+    comment = comment_service.find_by_id!(comment_id)
+    post = post_service.find!(comment.post.id) # checks implicit for public posts
```
Lint/UselessAssignment: Useless assignment to variable - `post`.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,84 @@
+
+        [alice, bob, eve].map {|user| LikeService.new(user).create_for_comment(comment.id)}
+        [alice, bob, eve].each do |user|
+        expect(
```
Layout/IndentationWidth: Use 2 (not 0) spaces for indentation.

diaspora-code-review commented on spec/services/like_service_spec.rb:
```diff
@@ -115,20 +175,84 @@
+
   end
-end
+end
```
Layout/TrailingEmptyLines: Final newline missing.

tclaus commented:
Found this API issues. 
rspec ./spec/integration/api/likes_controller_spec.rb:126 # Api::V1::LikesController#create with right post id fails in liking already liked post
rspec ./spec/integration/api/likes_controller_spec.rb:115 # Api::V1::LikesController#create with right post id succeeds in liking post
rspec ./spec/integration/api/comments_controller_spec.rb:147 # Api::V1::CommentsController#read valid post ID retrieves related comments

Can someone help here?

Flaburgan commented:
Hm yeah, the work on likes on comment is older than the work on the API. It means that: 
- The API doesn't support [liking comments](https://diaspora.github.io/api-documentation/) so this has to be added [documentation repo](https://github.com/diaspora/api-documentation). Maybe @jhass can give some tips here
- It is possible that some changes made broke the backend, you need to check that. The [federation part](https://github.com/diaspora/diaspora_federation) is supposed to be working (at least likes made on comments by friendica [are relayed](https://diaspora.github.io/diaspora_federation/entities/like.html)) but this has to be checked too.

tclaus commented:
Rspec only gives me the API errors, cucumber works locally on my machine. So I am on a point where I appreciate some help. 
Felling so close to a new feature.

Flaburgan commented:
@tclaus the current test failing doesn't look related to the API, it's an undefined in JS, you can probably fix it I think. Once done, I will do a review.

tclaus commented:
Thanks @Flaburgan - I am still on it, and I fixed some rspec errors locally, but there is still a schema issue left over. 
The JS error won't occur If I run cucumber on my local machine. (And I've rebased to latest develop)

diaspora-code-review commented on app/assets/javascripts/app/collections/comments.js:
```diff
@@ -12,12 +12,17 @@ app.collections.Comments = Backbone.Collection.extend({
   make : function(text) {
     var self = this;
-    var comment = new app.models.Comment({ "text": text });
+    var comment = new app.models.Comment({ "text": text },{ post: this.post});
```
- There should be no space after '{'
- There should be no space before '}'
- A space is required after ','.

diaspora-code-review commented on app/assets/javascripts/app/models/comment.js:
```diff
@@ -1,6 +1,16 @@
+  urlRoot: "/comments",
+
+  initialize: function(model, options) {
+    this.post =  options.post || this.collection.post;
```
Multiple spaces found before 'options'.

diaspora-code-review commented on app/services/translation_service.rb:
```diff
@@ -0,0 +1,17 @@
+
+  def translate_for_post(post)
+    {
+      translation: "translation for post with ID: " + post.text.to_s 
```
Layout/TrailingWhitespace: Trailing whitespace detected.

diaspora-code-review commented on app/services/translation_service.rb:
```diff
@@ -0,0 +1,17 @@
+  end
+
+# Task 1:  Make request to DeeplSDervice
+# Task 2: Make front end
```
Layout/CommentIndentation: Incorrect indentation detected (column 0 instead of 2).

diaspora-code-review commented on app/services/translation_service.rb:
```diff
@@ -0,0 +1,17 @@
+
+# Task 1:  Make request to DeeplSDervice
+# Task 2: Make front end
+
```
Layout/EmptyLinesAroundClassBody: Extra empty line detected at class body end.

diaspora-code-review commented on app/services/translation_service.rb:
```diff
@@ -0,0 +1,17 @@
+# Task 1:  Make request to DeeplSDervice
+# Task 2: Make front end
+
+end
```
Layout/TrailingEmptyLines: Final newline missing.

diaspora-code-review commented on spec/javascripts/jasmine_helpers/factory.js:
```diff
@@ -47,7 +47,8 @@ var factory = {
       "guid" : this.guid(),
       "id" : this.id.next(),
-      "text" : "This is a comment!"
+      "text" : "This is a comment!",
```
Extra space after key 'text'.

diaspora-code-review commented on spec/javascripts/jasmine_helpers/factory.js:
```diff
@@ -47,7 +47,8 @@ var factory = {
       "id" : this.id.next(),
-      "text" : "This is a comment!"
+      "text" : "This is a comment!",
+      "post" : this.post()
```
Extra space after key 'post'.

diaspora-code-review commented on spec/javascripts/jasmine_helpers/factory.js:
```diff
@@ -43,13 +43,15 @@ var factory = {
+      "text": "This is a comment!"
     };
-
+    
```
Trailing spaces not allowed.

diaspora-code-review commented on spec/javascripts/jasmine_helpers/factory.js:
```diff
@@ -43,13 +43,15 @@ var factory = {
     };
-
+    
+    overrides = overrides || {}
```
Missing semicolon.

denschub commented:
@tclaus, could you please run stylechecks locally on your machine **before** pushing a commit? The bot is meant as a safety-check if you miss something, not really as a tool to figure things out. :)

tclaus commented:
All checks have passed ❤️

Flaburgan commented:
So, thank you very much for your work @tclaus 
I did a "User point of view" review: I launched your branch locally and had a look.
First thing I saw is that there is a line break between the icon and the "1 J'aime":
![image](https://user-images.githubusercontent.com/930064/108899546-d59c1200-7618-11eb-90cb-71f29f0189d1.png)
Note that this is not the case for post. So you're missing some CSS rules here.
Another thing, not sure it should be part of that PR but maybe: the users don't have any notifications when their comment is liked. 
And last remark: those likes aren't displayed with the mobile interface, and it's not possible to like a comment there too. We have a long history of the mobile interface lagging behind in terms of features and @jhass is actually close to publishing a nice Android (and iOS) app, so you probably don't need to deal about that in this PR but I wanted to point it.
Apart from that, it looks like it works nicely, great work!

Also, I didn't test the federation part (as I only have my local test pod), but I could deploy that branch on diaspora-fr.org once you fix the CSS problem and I checked the code a bit.

tclaus commented:
@Flaburgan 
I fixed the CSS style, I also done some event handling issues. Looks better now. 

The mobile interface is next. 

For creating a notification a starting point for creating notifications would be nice - any tips here? 
For federation: No clue what's this meant about comment-likes. As the likes is federated in the Model, it may already solved? 
I would love to see this on a real website.

Flaburgan commented:
Yes that's better, there still is a small problem with the avatar size:

![image](https://user-images.githubusercontent.com/930064/109314723-00b97800-784a-11eb-8832-0afc7c79fdf5.png)

Flaburgan commented:
Also you have a commit about http -> https links in that branch, maybe you need to rebase on develop?

tclaus commented:
> Also you have a commit about http -> https links in that branch, maybe you need to rebase on develop?

No, that's dirt from another PR. Do you have an idea to remove these file from the commits? Do I have to reset everything and commit again?

On the weekend I will work on the other topics.

Flaburgan commented:
> > Also you have a commit about http -> https links in that branch, maybe you need to rebase on develop?
> 
> No, that's dirt from another PR. Do you have an idea to remove these file from the commits? Do I have to reset everything and commit again?
> 
> On the weekend I will work on the other topics.

Rebase on the develop branch and squash your commits, it will be gone.

tclaus committed:
set correct comment like avatar by removing size

diaspora-code-review commented on app/assets/stylesheets/comments.scss:
```diff
@@ -24,10 +24,7 @@
-      height: 35px;
-      width: 35px;
-    }
+   
```
Line contains trailing whitespace

tclaus committed:
Enabled likes-on-comments in mobile view

diaspora-code-review commented on app/helpers/mobile_helper.rb:
```diff
@@ -39,6 +39,21 @@ def mobile_like_icon(post)
   end
 
+  def mobile_like_comment_icon(comment)
+    puts "Mobile like on comment!"
```
Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.

diaspora-code-review commented on app/helpers/mobile_helper.rb:
```diff
@@ -39,6 +39,21 @@ def mobile_like_icon(post)
 
+  def mobile_like_comment_icon(comment)
+    puts "Mobile like on comment!"
+    if current_user && current_user.liked?(comment)
```
Style/SafeNavigation: Use safe navigation (`&.`) instead of checking if an object exists before calling the method.

diaspora-code-review commented on app/assets/stylesheets/mobile/comments.scss:
```diff
@@ -71,12 +71,21 @@
   }
 
+  .comment-stats {
+    @extend .post-stats;
```
Prefer using placeholder selectors (e.g. %some-placeholder) with @extend

diaspora-code-review commented on app/assets/stylesheets/mobile/comments.scss:
```diff
@@ -71,12 +71,21 @@
   }
+  
+  .comment-action {
+    @extend .post-action;
```
Prefer using placeholder selectors (e.g. %some-placeholder) with @extend

diaspora-code-review commented on app/assets/stylesheets/mobile/comments.scss:
```diff
@@ -71,12 +71,21 @@
 
     .disabled { color: $medium-gray; }
   }
+  
```
Line contains trailing whitespace

tclaus committed:
Adding eventhandler for Mobile views

diaspora-code-review commented on app/assets/javascripts/mobile/mobile_comments.js:
```diff
@@ -111,6 +111,8 @@
           bottomBarContainer.getCommentsContainer().find("time.timeago").timeago();
           bottomBarContainer.activate();
+          // Inform the comment action for new comments
+          $(".stream").trigger('comments.loaded');  
```
- Strings must use doublequote.
- Trailing spaces not allowed.

diaspora-code-review commented on app/assets/javascripts/mobile/mobile_post_actions.js:
```diff
@@ -3,6 +3,11 @@
       $(".reshare-action", ".stream").bind("tap click", this.onReshare);
+      // Add handler to newly loaded comments
+      var self = this;
+      $(".stream").bind("comments.loaded", function(evt){
```
- 'evt' is defined but never used
- Missing space before opening brace.

tclaus replied:
Sorry for these Style - warnings. I switched to VSCode recently and this 'saves' files on commit. Before this 'pronto' rund well locally. Thats why some style checks still are overseen.

SuperTux88 replied:
Save on commit sounds like an incredibly bad idea, what if you also want to test and run your changes locally before you commit, and then you would still see the old changes? :thinking: 

But I can recommend an automated pre-push hook, so pronto runs before you push and tells you the result (and aborts the push if needed): https://wiki.diasporafoundation.org/Styleguide#Perform_an_automatic_local_review_before_pushing_to_your_repo

tclaus replied:
Ah.. the '--unstaged' option in pronto may be the missing part here.

tclaus commented:
@Flaburgan I think, Mobile View works well now. I would appreciate if you can roll out this for testing on your machine.

Flaburgan commented:
> @Flaburgan I think, Mobile View works well now. I would appreciate if you can roll out this for testing on your machine.

Mobile version looks OK, great job. However, the design is still broken in Single Post View on desktop: 
![image](https://user-images.githubusercontent.com/930064/109430269-c8a86580-7a00-11eb-925a-7393c510bcee.png)

tclaus commented:
@Flaburgan 
With ca537e970fd9e62a3e0c7ff46d2d9053d0fdf14c the single-post view looks now like this: 

<img width="1240" alt="Bildschirmfoto 2021-03-01 um 09 06 23" src="https://user-images.githubusercontent.com/501326/109470753-21631700-7a70-11eb-9368-a45b156946d1.png">
<img width="1219" alt="Bildschirmfoto 2021-03-01 um 09 06 29" src="https://user-images.githubusercontent.com/501326/109470757-22944400-7a70-11eb-98c0-b285b1d33a80.png">

Flaburgan commented:
So, sorry to take that long to answer you. The result is better now, but I still feel like it can be more polished. Space used can be optimized. I'm thinking about thread with many comments, that raises the scrolling time.

This is how comments and likes look on facebook:
![image](https://user-images.githubusercontent.com/930064/111538742-590ad800-876d-11eb-8f5b-47057bb371be.png)

So basically, the action to like a comment and the likes are on the same row. Maybe we can even take the occasion to change that part to actually move to more touch friendly icons, as you saw for the single post view the "like" action already is the heart, like for the mobile view. This is basically what I proposed here: https://github.com/diaspora/diaspora/issues/7996

Tell me if you are interested to do those design changes, or if we prefer to focus this PR on the logical stuff and will do a second pass for the design.

tclaus commented:
Hi @Flaburgan - for me I am open for a threaded comments view, and also to slightly redesign the Like-Action and the like view.. but..
This will change much more - where to show the liked-people icons?  The comments itself should be shown with a bubbled- flex width.. etc. maybe we should use pen&paper first. 

What do you think to make smaller steps, instead of the  bigger 'one for all'? - Have a look at some long lasting PRs and discussions. Let us bring this like feature out, and start immediately the next ideas. Then threaded comments. 
What do you think? 

In the meantime, I deploy this to my (brand-new) pod. https://societas.online there this can be evaluated in a life scenario.

tclaus commented:
@Flaburgan Deployment on my test-pod is showing likes from friendica. So federation seems to work in incoming direction.

Flaburgan commented:
Did you solve your problem with Cucumber test btw? Here is a better place to discuss about it than in private message ;)

tclaus commented:
> Did you solve your problem with Cucumber test btw? Here is a better place to discuss about it than in private message ;)

The cucumber-Problem was not here, it was in #8217 (Multi Select Aspects in Mobile). On this I may need some help. 
Sorry, I have opens a bunch of PRs the last days. 

Here, everything seems to be OK, even tests.

jhass commented on app/assets/javascripts/app/models/post/like_interactions.js:
```diff
@@ -0,0 +1,57 @@
+// This class contains code extracted from interactions.js to factorize likes management between posts and comments
+
+app.models.Post.LikeInteractions = Backbone.Model.extend({
```
Nitpick: Any reason to nest this into the `Post` namespace? Given this applies to comments as well now that feels a little weird to me.

tclaus replied:
This Idea was inherited from Hank as initiator- what do you suggest?

jhass replied:
Just moving it to the top-level namespace perhaps?

tclaus replied:
Sounds good. Will work in it in the next days.

tclaus replied:
@jhass - ping? I've done some changes on the code as requested.

jhass reviewed (changes requested):
I'm sorry but I'm afraid I'll have to request a couple basic integration (cucumber) tests here.

tclaus committed:
Refacor files namespaces

tclaus commented:
Ping @jhass
I added tests and fixed code.  Can you approve it now and give a new target Milestone (8 maybe?)

goobertron commented:
Has anyone tested this on a pod with a large database? As I remember it, this feature was removed because it was blowing up database tables and slowing down pods. Is there any way of knowing whether this would still be a problem for larger pods?

tclaus commented:
> Has anyone tested this on a pod with a large database? As I remember it, this feature was removed because it was blowing up database tables and slowing down pods. Is there any way of knowing whether this would still be a problem for larger pods?

In https://github.com/diaspora/diaspora/issues/4920#issuecomment-146895248 a long discussion founds out, that at least some columns are not needed in the likes-table. 
The result was, to significantly sharing the space the likes-table consumes. 
So database size should be watched, but not a problem.

Flaburgan commented:
Just to inform you that I plan to work on the design here because I think it's the only thing which is missing.

cmrd-senya reviewed (commented):
The code generally looks good to me. I wanted to run the PR to see how it works and to test it, but there are merge conflicts and the old base branch doesn't work anymore with the fresh DB schema of diaspora, so I couldn't test it yet.

tclaus commented:
When I am back in my office, I rebase it and fix conflicts.

tclaus commented on app/services/like_service.rb:
```diff
@@ -5,11 +5,17 @@ def initialize(user=nil)
 
+  def create_for_comment(comment_id)
+    comment = comment_service.find!(comment_id)
+    post_service.find!(comment.post.id) # checks implicit for visible posts
```
I have checked this - this function is for "like on comment", the post_id is unknown in the execution path.

cmrd-senya replied:
Should work this way: 
```suggestion
    post_service.find!(comment.commentable_id) # checks implicit for visible posts
```
It could be a little better because it won't instantiate `Post` object just to get it's `id`. Might save some bits of resources, in case if there are no other calls to `post` for this `comment` object.

tclaus replied:
The commentable_id works here. Thanks for the idea.

tclaus commented:
@cmrd-senya I have rebased it.

tclaus commented:
> Just to inform you that I plan to work on the design here because I think it's the only thing which is missing.

With the discussions ongoing on discuss, I think no more work is needed here. (before releasing it)

Flaburgan commented:
> > Just to inform you that I plan to work on the design here because I think it's the only thing which is missing.
> 
> With the discussions ongoing on discuss, I think no more work is needed here. (before releasing it)

Last time I checked (a few weeks ago) there was still the problem with the avatar size on the single post view.

tclaus commented:
@Flaburgan Using now micro avatar sizes in single stream view.

Flaburgan commented:
Thank you, I'm in vacation until the 12th of August but I will have a look once I'll be back.

cmrd-senya reviewed (approved):
Looking pretty solid to me.

I have tested it locally, but I haven't tested how federation works. It should work anyway as the support from the federation layer was there before this PR and the wiring looking good to me.

P.S. My approval doesn't let it get merged anyway but I hope the fact that I reviewed it is still helpful.

Flaburgan commented:
In SPV avatars are missing the `margin-right: 3px` CSS rule:
![image](https://user-images.githubusercontent.com/930064/184496724-3e5e039b-547e-4576-a6be-5bc705327251.png)

In the stream:

![image](https://user-images.githubusercontent.com/930064/184496754-97f1b62a-b3a1-4a2e-8eb0-3ae99378b262.png)

Flaburgan commented:
Also, I would put a `margin: 0` on the `.media` element containing the likes, it's wasting space at the moment.

tclaus commented:
@Flaburgan the css is using heavily xpaths that makes it hard(er) to exactly set properties for SPV and stream view. Do you mind helping me with a suggestion and code snipped for the css change?

diaspora-code-review commented on app/assets/stylesheets/single-post-view.scss:
```diff
@@ -164,4 +164,20 @@
+  .likes {
+    .author-name {
+      margin-right: 3px;
+    }
```
Rule declaration should be followed by an empty line

SuperTux88 reviewed (changes requested):
So, I only had a "quick" look at this, so this is not a full review yet (I basically haven't looked at any code yet), but as I played a bit with it in the frontend, I still wanted to give some feedback.

So in my opinion the likes take up way to much vertical space. Even with only the "Like" link, there is a lot of whitespace below the link, as the container for the likes is already there empty but still takes space:

![image](https://user-images.githubusercontent.com/458548/186547713-db5ae8c4-1cf6-4a07-85a6-8e89a8cb1461.png) ![image](https://user-images.githubusercontent.com/458548/186547643-754490d9-0d04-4519-b971-706272457371.png)
(comparing current `develop` with this PR)

And if there are likes, there is even more white space below it, there is basically enough space for a whole second row of avatars:

![image](https://user-images.githubusercontent.com/458548/186553510-2a14d638-05b9-4a2c-9f09-2a25076690f0.png) ![image](https://user-images.githubusercontent.com/458548/186547948-328a4069-e3ec-4f15-97e7-75ba79cd7126.png)

(all screenshots above are from the stream view, then spacing is slightly less bad on the single post view, but it still needs a lot of vertical space there too)

I also don't like how much space this ![image](https://user-images.githubusercontent.com/458548/186548296-fe2cfd01-82ed-473f-a9ee-2609051af921.png) takes as soon as there is one like, maybe only show a compact text (text only, no heart icon) "1 person likes this", with less spacing around it? And then only show the heart icon when clicking on it when the avatars are shown? What do others think about this idea? :thinking: 

![image](https://user-images.githubusercontent.com/458548/186551261-06bb5e65-db4c-44e2-bc80-7b2e478c8b38.png)

Also this "1 Like" link is blue on the single post view, and also in a different font size. This should probably look like in the stream also on the single page view (gray with smaller fontsize):
![image](https://user-images.githubusercontent.com/458548/186551765-2bc78e60-2d15-44a6-bea4-048a5eb777ff.png)

And I noticed there are no notifications for likes on comments yet.

tclaus commented on app/presenters/comment_presenter.rb:
```diff
@@ -19,11 +20,25 @@ def as_api_response
+
+  def build_interactions_json
+    {
+      likes:       as_api(likes),
```
Hm - we do need this information to generate "like" or "unlike" buttons. For this a fetch has to be done. The correct link text is then done by the frontend. 
I do not have any idea if-  and how to change that.

SuperTux88 replied:
OH GOD, really? ... but why does this require ALL likes already and why are ALL likes then refetched anyway if they are already there? This isn't directed at you, it's more questioning why things were done that way in the past. But also it's not so much a problem so far for posts, as there are loaded 15 together at most, but for comments it could load hundreds of comments for a big post, so it becomes more of a problem now.

For the API I would just do what is also done for the posts API, as there is a `own_interaction_state` with booleans.

For the frontend I don't know how hard it is to also use such boolean? If it's too difficult to change the frontend to use a boolean instead of the list, we could just have the `likes` only containing the own like (as for rendering the likes there is a separate request fetching all likes again anyway), but this would be a pretty bad workaround, so I'm not sure how much I like it. But as the frontend is only an internal API and we will eventually replace it with a new frontend, it might be fine?

(And sorry, as written in the Review already, I didn't really look at the the code, so I didn't know the likes in the initial response is used for that, I only saw that the likes were already included in the initial response and then they are fetched again when the are shown, so I assumed the initial likes weren't used for anything)

tclaus commented:
> And I noticed there are no notifications for likes on comments yet.

How to create one? Could this be done after #8375 is merged?

diaspora-code-review commented on app/assets/stylesheets/comments.scss:
```diff
@@ -42,6 +39,10 @@
       padding-left: 3px;
     }
+
+    .media {
```
Selector should have depth of applicability no greater than 3, but was 4

diaspora-code-review commented on app/assets/stylesheets/comments.scss:
```diff
@@ -42,6 +39,10 @@
     }
+
+    .media {
+      margin-bottom: 0px;
```
`0px` should be written without units as `0`

diaspora-code-review commented on app/assets/stylesheets/single-post-view.scss:
```diff
@@ -164,4 +164,25 @@
+      margin: 0;
+    }
+
+    a.expand-likes {
```
Avoid qualifying class selectors with an element.

diaspora-code-review commented on app/assets/stylesheets/single-post-view.scss:
```diff
@@ -142,7 +142,7 @@
   .comments > .comment,
   .comment.new-comment-form-wrapper {
-    padding: 10px;
+    padding: 10px 10px 0 10px;
```
- Shorthands of length `4` are not allowed. Value was `10px 10px 0 10px`
- Shorthand form for property `padding` should be written more concisely as `10px 10px 0` instead of `10px 10px 0 10px`

SuperTux88 commented:
> How to create one? Could this be done after https://github.com/diaspora/diaspora/pull/8375 is merged?

You need a [notification class](https://github.com/diaspora/diaspora/blob/develop/app/models/notifications/liked.rb) and a [mail worker](https://github.com/diaspora/diaspora/blob/develop/app/workers/mail/liked.rb), a [mailer class](https://github.com/diaspora/diaspora/blob/develop/app/mailers/notification_mailers/liked.rb), and a [text](https://github.com/diaspora/diaspora/blob/develop/app/views/notifier/liked.text.erb) and [html](https://github.com/diaspora/diaspora/blob/develop/app/views/notifier/liked.html.haml) template (these are the current like-on-post files, so they need to be slightly altered for likes on comments).

But yes, I think it can be done after the other PR, as some of these classes change slightly. But it should be done in the same release as likes on comments get released.

tclaus committed:
In API dont fetch liked from all comments

tclaus committed:
Dont send all likes with list of comments

Due to historic reasons with a comment the list of all likes was sent to the frontend.
This is needed just to detect if one of the likes is current users like.
So if sending just the own like, the frontend can do it's job.

When the frontend is refactured in any way, post and comment like handling should be improved.

diaspora-code-review commented on app/assets/javascripts/app/helpers/handlebars-helpers.js:
```diff
@@ -129,4 +129,9 @@ Handlebars.registerHelper('aspectMembershipIndicator', function(contact,in_aspec
 });
+
+// Returns a value that defines to show a heart before like text or avatar list
+Handlebars.registerHelper('shouldShowHeart', function(likePresenter) {
```
Strings must use doublequote.

cmrd-senya commented:
Just my 5 cents about the UI/UX. While I can agree that some things may be improved here, I don't know if it's worth being polished in this PR. There is value in breaking things into smaller pieces. I'm sure Thorsten wants to get this merged and move on. We may decide whether we want to release things this way or we want to block the 1.0.0 release before we improve the designs of the likes for comments. But since it is "develop" branch I don't think we should block merging this feature just because of the design.

I think it's up to Thorsten -- if he says that he's ready to improve the UI/UX in the scope of this PR it is fine. But this PR is old enough and if it is hard for Thorsten to change it I'd prefer to ship it and go on a next circle. It's important for all of us to feel that there is some progress in what we're doing, so @SuperTux88  I suggest to focus on what's absolutely necessary to get fixed to merge this pull request.

tclaus commented:
@cmrd-senya I think also to get this done and make (mini-)improvements it with new and smaller PRs. Supertux found some issues to fix and also sent me privately some Ideas to improve the design, which I have done mostly in the last commit.
But js code and html templates are shared with the parts for the likes for posts, any design change is not an easy task. 
I'd love to see this PR finally merged, whoever wants to make design changes or improvements can send me a patch file, or can create directly a commit. 
Or just opens a new PR.

denschub commented:
> While I can agree that some things may be improved here, I don't know if it's worth being polished in this PR. 

There are two problems with this approach. We've merged features before where the contributors said "yeah don't worry I'll file follow-up PRs to fix all the issues" - and then they just... didn't. Now, I fully trust that Thorsten will not suddenly disappear, but I think it's important for a project to have *consistent* accept criteria for everyone, regardless of if you're an established core contributor, or someone who just submitted their first PR. Otherwise, projects can feel incredibly arbitrary, and it's easy to build some kind of "inside culture". Even if we would agree on merging this PR with it's current UX/UI, the test suite is still completely on fire, and there is no chance this will land in `develop` with burning tests.

The second problem is that there are pods running `develop` sources. Geraspora, for example, is one of them - and I did that to allow for better pre-release testing of code with actual users. If we imagine that this PR would be merged as-is, I'd have to hold off on deploying new changes until that follow-up PR is merged, because my users will start pitchforking me if stuff changes every other week. Now, this is kinda my problem since I'm the one who decided to run a production pod on development sources, but I felt like this is a point worth adding.

At the same time, I get that dealing with a PR this large and this old is a pain. A thing we could do is merge this PR into a new feature branch, like `feature/likes-on-comments`. That would get rid of this PR, open the slate for new PRs that improve the implementation in "smaller" chunks, without affecting the mainline branches. If we do that, we should file a "meta-bug" that tracks the remaining steps for that feature branch to be merged into `develop`. We'd have to spend a bit of effort in rebasing the feature branch occasionally, which is nothing Thorsten could do given the lack of write-access to the repo, but that's not a lot of work, and I can volunteer to do that.

SuperTux88 commented:
Yes, it doesn't need to be absolutely polished, and I'm fine with moving some stuff to later PRs, for example the notifications can be done later, because without this PR you also don't get notifications, while the likes already work in the background through federation, and this PR makes them visible, but no notifications yet, and notifications can then be done later to complete the feature at the end.

But we have pods running on the `develop` branch with a lot of users on it, and we should still at least polish it as much that it's not making everything a lot worse and looks somehow OKish. And the initial state of how it looked was just really bad and added a lot of noise and wasted space everywhere (in my opinion), and you can just optimize it a little with some "quick" CSS changes so it looks somewhat consistent.

I know the `develop` branch isn't a finished release and there is some risk running it and things are allowed to break, but we usually don't merge stuff that is knowingly broken and we also shouldn't merge stuff that makes things worse (and for example missing notifications doesn't make things worse, you already don't have notifications for likes on comments now, even while the likes themself already are there ... but an inconsistent UI would make things worse in my opinion).

So I'm not asking for a complete new concept of likes and a new UI for everything, but it should at least look somehow consistent (font-sizes, colors, spacing), and as this PR is mostly about bringing the likes on comments back to the frontend, I think it isn't asked too much to focus a bit on how it looks, as this is in my opinion in the scope of this PR (while for example notifications can be done in a separate PR).

SuperTux88 commented:
> A thing we could do is merge this PR into a new feature branch, like `feature/likes-on-comments`.

I just saw the comment from Dennis after I submitted mine, and yes, merging this to a feature-branch might be a solution if Thorsten doesn't want to work on this anymore, or if somebody wants to work on fixing some of the style issues, they can also send a PR to Thorstens branch (but please coordinate this with Thorsten, so not multiple people start working on this now ;) ).

> so @SuperTux88 I suggest to focus on what's absolutely necessary to get fixed to merge this pull request.

But to answer that question: As this PR is about the frontend of likes on comments, I do think that having at least a somewhat consistent UI is absolutely necessary to get this merged to `develop`.

diaspora-code-review commented on app/controllers/comments_controller.rb:
```diff
@@ -49,7 +49,7 @@ def new
     comments = comment_service.find_for_post(params[:post_id])
     respond_with do |format|
-      format.json { render json: CommentPresenter.as_collection(comments), status: 200 }
+      format.json { render json: CommentPresenter.as_collection(comments, :as_json, current_user), status: 200 }
```
Rails/HttpStatus: Prefer `:ok` over `200` to define HTTP status code.

tclaus commented:
I did not read the conversation between Dennis and SuperTux while I was fixing some issues. 
The current state ist, I did some minor changes (hopefully) in the right way in terms of design, but I would appreciate help in doing the CSS Styles.

tclaus commented:
BTW: I like the Idea of a separate feature branch and working on design in a separate Branch.

Flaburgan commented:
> I did not read the conversation between Dennis and SuperTux while I was fixing some issues. The current state ist, I did some minor changes (hopefully) in the right way in terms of design, but I would appreciate help in doing the CSS Styles.

I can help you with the CSS if we know what we want.

SuperTux88 commented:
> if we know what we want

My requirements are basically:
* somewhat consistent spacing/indented (above and below the text/links/avarats)
* have it consistent between stream view and single post view (same text size for links and texts, same colors)

Looking at the commit message (I didn't have a look at any code again, or tested it locally), Thorsten already implemented some of my suggestions, of having it as a text only link (without the heart) when the avatars aren't shown (to reduce visual clutter), so something like this:
![](https://user-images.githubusercontent.com/458548/186551261-06bb5e65-db4c-44e2-bc80-7b2e478c8b38.png)

And then only show the heart icon when the avatars are shown.

But as said, I haven't looked at how it now looks with the changes Thorsten did, so I can't tell what exactly is still needed to be fixed CSS wise (or if anything is needed at all), so you would need to check the current version yourself.

I will have a more detailed look at the rest of this PR hopefully soon and then we can decide if we want to merge this to a feature branch or to `develop`, but if you want to already help out Thorsten before that, you can just create a PR to his branch or send him a patch file.

SuperTux88 committed:
Fix styling and spacing for likes on comments

SuperTux88 committed:
Only show heart icon when avatars are shown for comment likes

SuperTux88 commented:
@tclaus I just force pushed to this PR with the following changes:
* Rebased on current `develop` (mainly to have the latest migrations so it's compatible with my database)
* Squashed https://github.com/diaspora/diaspora/pull/8203/commits/714821930e28a397dcd80c55c926cc8740a298c3 and https://github.com/diaspora/diaspora/pull/8203/commits/e1c71b66e101b5df63c7743a5fdd8547dfb77f37 to where they belong to reduce numbers of commits.
* Refactored the styling on desktop to fix the uneven spacing and indentation and different styles between stream and single-post-view. I tried to re-use as much existing styles (and styles shared between both views) to get a consistent styling everywhere. It might not be perfect yet, but it's as far as I'm happy with how it looks now. I did my style changes in my own commit (https://github.com/diaspora/diaspora/pull/8203/commits/a389dc87506ee458685a22ba0215044ba8be5afd) and removed https://github.com/diaspora/diaspora/pull/8203/commits/739acfd04d6bac085acf011a7820db1dc9654ec6, https://github.com/diaspora/diaspora/pull/8203/commits/0f88b91a8b9d938cbb424eeb6aaeac97b0931faa and https://github.com/diaspora/diaspora/pull/8203/commits/3b02cefa14c79ecc1cd3fc70c4df674b652e4fcb as they were 90% obsolete and replaced by my own code anyway and again to reduce number of commits with stuff that belongs together.
* I refactored the hiding of the heart icon when no avatars are shown to a CSS-solution (adding my won commit https://github.com/diaspora/diaspora/pull/8203/commits/9b682e22da59713aee8e62b508f2cc277c9f7f41, replacing https://github.com/diaspora/diaspora/pull/8203/commits/ad575ecd125db478dab9f174a4dd4e2d8f14a03c and also refactored and integrated https://github.com/diaspora/diaspora/pull/8203/commits/f57b2c5f62657f6f8710886a9c089ed8f2b2bc07) as I think that's simpler. I also fixed the broken cucumber test related to the hidden heart icon.

What is still to do:
* I still didn't really look at the rest of the code yet as I only focused on how the fronted looks. I plan to hopefully work on that soon.
* I noticed that the styling on mobile is also broken and I didn't fix that yet, but plan to also work on that.

tclaus commented:
@SuperTux88 Any progress here?

SuperTux88 commented:
Sorry, I didn't have any time to work on diaspora recently, and that also doesn't change in the following few weeks :disappointed:
but I'll hopefully be back in a few weeks.

memo-567 commented:
@SuperTux88 

Is there any news for this PR, or is it being forgotten because some admins, who know how to do it, have installed it manually by now?

tclaus commented:
> 

There seems to be some minor styling issues. I don't habe time to get into this, so I asked the admins here to fix it if the issues are major. 
As a podmin admin feel free to simply merge this PR into your code base. It is running on some machines well.

memo-567 commented:
@tclaus 

If there are only 

> some minor styling issues

should it be possible for the experienced core development team to correct this, or is there no longer any interest in further development?

tclaus commented:
@memo-567 I am sure this feature will find its way to the main branches. Development focussed now on next major (0.8 / 1.0) and migration issues. There is simply no time left in the developers spare time for this project. 
In my case, I have become a father. So time is critical for me.

memo-567 commented:
@tclaus 

Then I still hope for the best, after all, the beginning of your good work on this PR started two years ago, and I wish you a nice time as a father 😀

Flaburgan commented:
Looking at the last message by @SuperTux88 https://github.com/diaspora/diaspora/pull/8203#issuecomment-1312923795 there are two points left:
- Review of the logic -> This has to be done by one of the core dev, SuperTux, @denschub or @jhass 
- Fix the styling in the mobile version -> This I could do it, although I will leave in vacations very soon and I probably won't have time before that, so it may have to wait until end of August. I will do what I can, because I would really like to see this integrated in the next major version, mostly because there are not that many users facing changes and this one is waited for a long time.

Flaburgan committed:
Improve styling for mobile comments

Flaburgan commented:
I had a look at the style for mobile. I was not very creative and simply styled it the way it is for posts. I made each comment a box, it helps to understand the limits better. I always wondered why they were on a grey background, making it hard to read especially outside (where you can often be on mobile).

**Before / After**
![Capture d’écran 2023-10-17 à 22 50 07](https://github.com/diaspora/diaspora/assets/930064/aeaebc97-4a76-48d8-8d3d-2552e4bc10ef) ![Capture d’écran 2023-10-17 à 22 46 21](https://github.com/diaspora/diaspora/assets/930064/0e023c9d-8821-4e03-9187-9a088283ab4f)

Some cleaning of all of this CSS is definitely needed but I did not do big changes here, as we will probably ditch this front-end and especially this mobile version at some point.

I also rebased the branch on the last develop, this is working fine and is currently running on diaspora-fr in production.
I don't think I have the permission to push on @tclaus branch and because of the rebase it's not meaningful to open a merge request on his branch, so here is my commit: https://github.com/diaspora/diaspora/commit/94b67d27e81de3d9fd8d816964e78b2bba08161a

@tclaus if you can rebase your branch I will then open a MR, or you can cherry-pick it.

tclaus commented:
Hi Fla. feels good to see any progress here. I am still very busy raising my little son. But I think I can rebase today evening or tomorrow.  Today  I am sick and need to stay in bed. Btw: hasn’t Supertux not created a new branch in the diaspora repo?Am 17.10.2023 um 22:59 schrieb Fla ***@***.***>:﻿
I had a look at the style for mobile. I was not very creative and simply styled it the way it is for posts. I made each comment a box, it helps to understand the limits better. I always wondered why they were on a grey background, making it hard to read especially outside (where you can often be on mobile).
Before / After
 
Some cleaning of all of this CSS is definitely needed but I did not do big changes here, as we will probably ditch this front-end and especially this mobile version at some point.
I also rebased the branch on the last develop, this is working fine and is currently running on diaspora-fr in production.
I don't think I have the permission to push on @tclaus branch and because of the rebase it's not meaningful to open a merge request on his branch, so here is my commit: 94b67d2
@tclaus if you can rebase your branch I will then open a MR, or you can cherry-pick it.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Flaburgan commented:
@tclaus thanks, rebasing should be very straight forward, I had no conflicts. I will then open a MR on your branch.

@SuperTux88 did not need to do that, as he is a contributor of diaspora/diaspora he was able to push to your branch directly.

denschub commented:
> I was not very creative and simply styled it the way it is for posts.

I really like that - in fact, I like it more than the solution on the desktop/responsive version. 🙃 I like how the brighter background separates all comments even more. If you can fix the weirdly placed counter numbers, I'd be very happy with that. :)

> rebasing should be very straight forward, I had no conflicts. I will then open a MR on your branch.

Thorsten could add you as a contributor to his fork. Alternatively, you could just push his branch into your fork and work from there. We can always cherry-pick/merge later.

Flaburgan committed:
Fix pronto in Ruby

Flaburgan commented:
So thank you @tclaus I was able to push to your branch.

@denschub I am glad you like it. In the future I guess we can apply the same styling to the desktop version.
I improved a bit the alignment (the real solution would be to use flexbox but I don't want to go through a full refactor) and I also fixed the border-radius at the bottom. Here is how it looks like now:

![Capture d’écran 2023-10-19 à 10 41 36](https://github.com/diaspora/diaspora/assets/930064/fcd4aa4e-8e22-4afa-85d5-7375fe33d46a)

I also fixed most of the pronto errors. I guess it is ready for review.

SuperTux88 committed:
Fix API v1 schema for likes on comments and add test

SuperTux88 committed:
Fix participations for likes on comments in the backend

When liking a comment, the post also gets a participation, and if all
likes/comments get removed again, the participation also gets removed
again.

The only thing still not working properly is the frontend, but that is
already broken when unliking a post. So it shows an invalids state in
the frontend when unliking the post/comment.

SuperTux88 committed:
Fix comments not being lazy loaded in streams anymore

SuperTux88 committed:
Add notifications for likes on comments

SuperTux88 commented:
OK, I reviewed now everything and fixed all the things I found myself, as @tclaus doesn't have time at the moment. I now just need somebody who reviews my commits (the ones I just added, and the ones from a year ago)

> Edit: Jasmine tests are green locally, both with Firefox and Chrome... I will investigate.

@Flaburgan you broke that comments were lazy-loaded in the stream, which the test was testing for. And you didn't regenerate fixtures locally, so jasmine was still using the old fixtures where the comments were still lazy-loaded. I also fixed that so comments are laxy-loaded in the stream again.

SuperTux88 reviewed (approved):
Approving the commits and changes not made by myself ;)

denschub commented on lib/schemas/api_v1.json:
```diff
@@ -105,9 +105,11 @@
-              "likes_count" : { "type": "integer"}
-            }
+              "liked" : { "type": "boolean" },
+              "likes_count" : { "type": "integer" }
```
We show the people who liked a comment in the frontend. Is there a reason not to show a list of likees in the API as well?

SuperTux88 replied:
I think the reason is performance (as there could be quite many likes), there is a separate endpoint to get the complete list with all the information about the people who liked it. I assume fetching all the details about the people who liked it would just be too much when just fetching a list of posts/comments.

denschub replied:
> there is a separate endpoint to get the complete list with all the information about the people who liked it.

Okay, I have to admit, I 100% missed this. This is perfect, then, I'll mark this as resolved.

skyschub reviewed (approved):
This PR still has five Pronto Ruby violations:

- app/controllers/likes_controller.rb:31 W: Metrics/AbcSize: Assignment Branch Condition size for create is too high. [<2, 22, 3> 22.29/20]
- app/controllers/notifications_controller.rb:11 W: Metrics/AbcSize: Assignment Branch Condition size for index is too high. [<18, 51, 10> 55/20]
- app/controllers/notifications_controller.rb:11 W: Metrics/MethodLength: Method has too many lines. [29/20]
- app/controllers/notifications_controller.rb:70 W: Metrics/AbcSize: Assignment Branch Condition size for read_all is too high. [<4, 32, 4> 32.5/20]
- app/helpers/notifications_helper.rb:19 W: Metrics/AbcSize: Assignment Branch Condition size for object_link is too high. [<3, 25, 7> 26.13/20]

as well as one SCSS violation:


- app/assets/stylesheets/mobile/comments.scss:131 W: Selector should have depth of applicability no greater than 3, but was 4

If we don't want to fix them, maybe we should add disable comments for those? Although I'm also not opposed to just ignoring them, but I felt like this is worth highlighting.

Either way, I have one question, but I'm approving anyway as I don't see this as blocking.

SuperTux88 commented:
> If we don't want to fix them, maybe we should add disable comments for those?

I didn't want to add disable-comments for `AbcSize` and `MethodLength`, as it's not that "I don't want to fix them", more a "I don't want to fix them **NOW**". Because maybe these methods should be refactored (so I didn't want to just disable the warning now) or maybe we want to raise the limits globally for these rules if we are fine with that complexity?

As these violations were already there before, they just triggered now because I moved some minor things around, but otherwise didn't really touch these methods. And I also didn't want to refactor these methods in the scope of this PR, as it's already complex enough. So I just wanted to ignore them for now in this PR.

I'm not sure about the violation in the CSS, if we should change that or put a comment there or just also ignore it :man_shrugging:

denschub commented:
Let's ignore them, then. I'm fine with that, just wanted to raise that because I noticed. :)
